daemon/chaintopology.c

```diff
@@ -279,19 +279,19 @@ static void destroy_outgoing_tx(struct outgoing_tx *otx)
 void broadcast_tx(struct peer *peer, const struct bitcoin_tx *tx)
 {
 	struct outgoing_tx *otx = tal(peer, struct outgoing_tx);
-	char **txs = tal_arr(peer->dstate, char *, 1);
-	u8 *rawtx;
+	char *hextx;
 
 	otx->tx = tal_steal(otx, tx);
+	otx->peer = peer;
 	bitcoin_txid(otx->tx, &otx->txid);
 	list_add_tail(&peer->outgoing_txs, &otx->list);
 	tal_add_destructor(otx, destroy_outgoing_tx);
 
 	log_add_struct(peer->log, " (tx %s)", struct sha256_double, &otx->txid);
 
-	rawtx = linearize_tx(txs, otx->tx);
-	txs[0] = tal_hexstr(txs, rawtx, tal_count(rawtx));
-	bitcoind_sendrawtx(peer->dstate, txs[0], try_broadcast, txs);
+	otx->rawtx = linearize_tx(otx, otx->tx);
+	hextx = tal_hexstr(otx->rawtx, otx->rawtx, tal_count(otx->rawtx));
+	bitcoind_sendrawtx(peer->dstate, hextx, broadcast_tx_complete, otx);
```

Comment:
This is pretty ugly; the caller knows whether the failure is important or not, so it should hand the callback through broadcast_tx.  In particular, if the callback is NULL do the old behavior (or allow process_sendrawtx to take NULL and do nothing in that case), otherwise wire it up for the caller to do something.

I think the only caller we want to change is state_single(), which is where it does the funding tx.  (I have a patch lying around to clean that up, but it's a can of worms).